client/src/app/videos/+video-watch/video-watch.component.ts

```diff
@@ -323,6 +325,10 @@ export class VideoWatchComponent implements OnInit, OnDestroy {
     return this.video && this.video.scheduledUpdate !== undefined
   }
 
+  onRecommendations (videos: Video[]) {
+    this.nextVideoUrl = videos[0].uuid
```

Comment:
check the length of `videos` before assigning the first element + I fear that we'll have an infinite autoplay loop 

If think we should keep in memory the previous videos that we played to not play them again

Reply:
>an infinite autoplay loop

if the recommendation service returns always the same video, it will be an endless loop.
But how could I prevent this? Just remembering the last could resolve in a 3 way loop.

And shouldn't this rather be solved in the recommendation service?

Reply:
> if the recommendation service returns always the same video

Not only. If the service returns recently added videos, and you are watching the latest upload video you'll have (in the `[0]` index) the second last one uploaded video. Then you autowatch this video, and the service will return the latest uploaded video that you'll re-watch etc etc

> And shouldn't this rather be solved in the recommendation service?

It would be a solution :+1: